rcl/include/rcl/localhost.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__LOCALHOST_H
+#define RCL__LOCALHOST_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include "rcl/types.h"
+#include "rcl/visibility_control.h"
+
+extern const char * RCL_LOCALHOST_ENV_VAR;
```

Comment:
consider:
```suggestion
extern const char * const RCL_LOCALHOST_ENV_VAR;
```

Reply:
does it have to be extern here? any plan to use this value in other files as globally in the future? i do not see that is coming since there is a API provided to check if localhost is enabled or not on this.

Reply:
Right now is not used by anyone outside the localhost.c file. I made it extern to avoid copies using static and to fulfill [dirk's suggestion](https://github.com/ros2/rmw/pull/190#discussion_r334059644).
@dirk-thomas , any thoughts about keeping this variable in this context only?

Reply:
There is no point in hiding the definition of the name of the env var. If we do that people which want to use that will just copy-n-paste the name of the env var which is strictly worse. So keeping the name public is desired.

Reply:
(If we would do the same for `ROS_DOMAIN_ID` it would actually avoid it being define in three different source files.)